### ISSUE TITLE
Use with vcpkg

### ISSUE THREAD

**Issue by autoantwort** [Naios/idle#3]
This projects is very cool and I want to use it in my own project, but I use vcpkg for all my dependencies. I would also create a PR to add this lib to vcpkg, but I have found the following statement:
> Idle uses custom CMake scripts to download and set up its source dependencies automatically.

which is forbidden for vcpkg packages. Is there an option to disable this and is there a list of required dependencies?

**Comment by Naios** [comment]
I don't think support for vcpkg can be added. The project uses specific versions/revisions of its dependencies.
The custom dependency management was explicitly designed to support features, like version pinning, that vcpkg doesn't support.
You can try your luck by removing calls to `idle_enable_packages()`. The packages are then searched via default `find_package` cmake calls, that could be eventually provided by vcpkg as well.
Although I don't think that vcpkg is neccessary because the project is trivial to build and it creates an importable installation.

**Comment by Naios** [comment]
EDIT: Maybe the project is missing an option to pass `CMAKE_TOOLCHAIN_FILE` to the build directory initialization, which is probably required for vcpkg as stated by https://vcpkg.readthedocs.io/en/latest/examples/installing-and-using-packages/.

**Comment by autoantwort** [comment]
> The project uses specific versions/revisions of its dependencies.

Yeah I see. Like https://github.com/Naios/replxx/commit/66845b986c2c95b45783fbf6581eedad2a2a2bff, so you use modified versions of the dependencies. Yes that would prevent usage with any package manager. 

> Although I don't think that vcpkg is neccessary because the project is trivial to build with the provided buildsystem files and it creates an importable installation.

Not as easy as adding an entry `idle` to my vcpkg.json file and I am done and don't have to do an manual installation step for every configuration/system :)

> EDIT: ...

If I want to use vcpkg dependencies in the services this is probably true, otherwise this is not needed (Currently I don't do that).

**Comment by Naios** [comment]
All dependencies except replxx are the upstream versions, for replxx there are two outstanding PRs that contribute the modifications made on my fork back:

- https://github.com/AmokHuginnsson/replxx/pull/119
- https://github.com/AmokHuginnsson/replxx/pull/118

Overall the project uses 20 dependencies and it would be difficult and hindering to maintain a project with the fixed versions provided by vcpkg, which I also cannot modify if required.

**Comment by autoantwort** [comment]
Ok then I will wait until these PRs are merged. 

> Overall the project uses 20 dependencies and it would be difficult and hindering to maintain a project with the fixed versions provided by vcpkg, which I also cannot modify if required.

Yeah you don't have to care about it, it is the job of the vcpkg maintainers to provide the dependencies in the right versions. But when you say you always use upstream versions I think that shouldn't be an issue.

**Comment by Naios** [comment]
To follow up on this, I cannot guarantee to always depend on the upstream versions because if needed to progress (like in the current case of replxx), it takes too long to wait on the maintainers of the dependencies to include possible fixes.

I think the best way to accomplish providing *Idle* from within vcpkg is to distribute an amalgamation of the privately (non-leaking) dependencies and the source code itself. Then the few public dependencies could be polyfilled by vcpkg.
Therefore you only would have to download the amalgamation from within vcpkg

**Comment by Naios** [comment]
Since there was no further activity on this issue I'm marking it as resolved.
If you encounter any new bugs or questions open a new issue please.
